bryce-market-maker.md

```diff
@@ -0,0 +1,122 @@
+2. a more reliable timeseries based layer we can use to turn in tax information and have our quants do analysis of our profits and maybe tweak our algortihm accordingly
+
+
+
+# Initial Thoughts
+We need a very fast price comparison algorithm that also is greedy by nature. We want to grab the first stock that will replace our current stock.
+Another concern we might want to raise is how do we index stocks and if we see a pattern of increase do we want to buy a large chunk of shares? What
+can we do programatically to get larger than 1% of profits? Do we want to hold shares of every stock? or just select stocks? Do we want to buy stocks
+at a higher volume?
+
+# Step 2: Create High Level Design
+1. Web Server Layer responsible for serving orders to our frontend
+2. UI layer stateless retrieving data explicitly from the server
+3. Customer Database layer (KV store)
+4. Persistence layer(timeseries based datastore)
+
+
+
+
+The web server needs to have 4 router compoents
+
+1. Buy/Sell customer api responsible for reccomending stocks, aquiring a big enough stake for that trading day and executing the buy and sell orders.
+2. Authentication. All information should only be accessed through a user account. 
+3. transaction log. A durable history of all transactions our company has made.
+4. Historical Stock Index for all stocks we trade.
+5. Selection algorithm
+
+The UI Layer just needs a couple sections
+1. Search for stock
+2. Query stocks available to buy as well as a graph of price history.
+
+
+lets go over the scaling concerns of each one of our web server components.
+
+
+### Buy sell Layer
+Here we need to have a KV layer in which we serve a stock and its price on our market maker to our customer.
+A stock object will look like
+
+```python
+@dataclass
+class Stock:
+  ticker: str
+  price: int
+  timestamp: datetime
+```
+For each stock we will do two things with it
+
+#### View Concerns
+Every time the price of a stock updates we can have a new cell price. The KV store will have a key for each stock that we are selling as well as a value. 
+We will update this value every epoch second so we will want to have fast access to this value. Hash index is perfect for this. 
+this way accessing the value of a stock that is queried is O(1) average case and O(log(n)) worst case. 
+#### Buy Concerns
+We only want to buy a stock when it has dipped in value then sell it at the next spike. So we want to check the existing key value store and see if our target stocks are at a low price.
+Maybe some sort of target price can be generated where we also store the original buy price and only sell when the stock has gone up by 1%
+```python
+if buy stonk:
+	transfer stock to new owner 
+	then:
+      if successful
+		update our transaction ledger
+      else;
+		respond with error message
+```
+#### Sell Concerns
+We want to sell expensive shares and try to scam our customers. 
```

Comment:
actually we're the customer. Market makers buy and sell a spread, so we don't really want to hold much stock that's volatile, otherwise we'll lose money.

Reply:
I must have misunderstood